src/PrestaShopBundle/Controller/Admin/Sell/Catalog/CategoryController.php

```diff
@@ -82,31 +100,37 @@ class CategoryController extends FrameworkBundleAdminController
      * @return Response
      */
     #[AdminSecurity("is_granted('read', request.get('_legacy_controller')) || is_granted('update', request.get('_legacy_controller')) || is_granted('create', request.get('_legacy_controller')) || is_granted('delete', request.get('_legacy_controller'))", message: 'You do not have permission to list this.')]
-    public function indexAction(Request $request, CategoryFilters $filters)
-    {
-        $categoriesKpiFactory = $this->get('prestashop.core.kpi_row.factory.categories');
-
+    public function indexAction(
+        Request $request,
+        CategoryFilters $filters,
+        #[Autowire(service: 'prestashop.core.kpi_row.factory.categories')]
+        KpiRowFactoryInterface $categoriesKpiFactory,
+        #[Autowire(service: 'prestashop.adapter.category.category_view_data_provider')]
+        CategoryViewDataProvider $categoryViewDataProvider,
+        #[Autowire(service: 'prestashop.core.grid.factory.category_decorator')]
+        GridFactoryInterface $categoryGridFactory,
+        EmployeeContext $employeeContext,
+        MultistoreContextCheckerInterface $multistoreContextChecker,
```

Comment:
Same question I wonder if this should be in the parent class with an associated helper method